main.go

```diff
@@ -19,39 +19,33 @@ package main
 import (
 	"cloud.google.com/go/pubsub"
 	"context"
+	"fmt"
 	"github.com/numaproj-contrib/gcp-pub-sub-source-go/pkg/pubsubsource"
 	"github.com/numaproj/numaflow-go/pkg/sourcer"
 	"log"
 	"os"
 )
 
-func ensureTopicAndSubscription(ctx context.Context, client *pubsub.Client, topicID, subID string) (error, *pubsub.Subscription) {
+// ensureTopicAndSubscription checks if the specified topic and subscription exist.
+// It returns an error if either the topic or the subscription doesn't exist.
+func ensureTopicAndSubscription(ctx context.Context, client *pubsub.Client, topicID, subID string) (*pubsub.Subscription, error) {
```

Comment:
Please change the name of this method to "getSubscription" and update comments accordingly